metric/doc.go

```diff
@@ -31,23 +31,23 @@ part of a system is being measured.
 Instruments are categorized as Synchronous or Asynchronous and independently
 as Adding or Grouping. Synchronous instruments are called by the user with a
 Context. Asynchronous instruments are called by the SDK during collection.
-Additive instruments are semantically intended for capturing a sum. Grouping
```

Comment:
From the specification it looks like `additive` is still used over `adding`.

Reply:
Hmm. I thought the terms had fallen out of use. While I think they are a good classification system for instruments, it isn't necessary to have such a classification for most purposes.

In any case, I think "adding" was preferred to "additive" because if you look those words up in the dictionary, the form of "additive" we might use is much further from the common usage. 

Whatever it is, this distinction is meant as a way to separate the counter instruments from histogram and gauge instruments, because they have different meaning when you erase an attribute. See here: https://github.com/open-telemetry/opentelemetry-specification/issues/625

Also https://github.com/open-telemetry/opentelemetry-go/blob/e209ee7585bcb307539a4d58ff16d39b640f58f4/metric/sdkapi/instrumentkind.go#L55

I think it would be good to overhaul this documentation -- maybe remove the term.

(An aside: in today's SIG we discussed how I would **very much** like to remove some of the complications in the metric SDK processor/basic code, mainly this `Subtract()` implementation that is only needed when converting Cumulative to Delta which I spoke against supporting in the SDKs. As the argument goes, consumers of delta-only counters would prefer to see Cumulative sums transformed to Gauge data points.)